marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/MarathonPlugin.kt

```diff
@@ -1,136 +1,112 @@
+                        analyticsConfiguration.set(extension.analyticsConfiguration)
+                        poolingStrategy.set(extension.poolingStrategy)
+                        shardingStrategy.set(extension.shardingStrategy)
+                        sortingStrategy.set(extension.sortingStrategy)
+                        batchingStrategy.set(extension.batchingStrategy)
+                        flakinessStrategy.set(extension.flakinessStrategy)
+                        retryStrategy.set(extension.retryStrategy)
+                        filteringConfiguration.set(extension.filteringConfiguration)
+
+                        baseOutputDir.set(extension.baseOutputDir)
+                        ignoreFailures.set(extension.ignoreFailures)
+                        codeCoverageEnabled.set(extension.isCodeCoverageEnabled)
+                        fallbackToScreenshots.set(extension.fallbackToScreenshots)
+                        strictMode.set(extension.strictMode)
+                        uncompletedTestRetryQuota.set(extension.uncompletedTestRetryQuota)
+
+                        testClassRegexes.set(extension.testClassRegexes)
+                        includeSerialRegexes.set(extension.includeSerialRegexes)
+                        excludeSerialRegexes.set(extension.excludeSerialRegexes)
+
+                        testBatchTimeoutMillis.set(extension.testBatchTimeoutMillis)
+                        testOutputTimeoutMillis.set(extension.testOutputTimeoutMillis)
+                        debug.set(extension.debug)
+
+                        screenRecordingPolicy.set(extension.screenRecordingPolicy)
+
+                        applicationPmClear.set(extension.applicationPmClear)
+                        testApplicationPmClear.set(extension.testApplicationPmClear)
+                        adbInitTimeout.set(extension.adbInitTimeout)
+                        installOptions.set(extension.installOptions)
+                        serialStrategy.set(extension.serialStrategy)
+
+                        screenRecordConfiguration.set(extension.screenRecordConfiguration)
+
+                        analyticsTracking.set(extension.analyticsTracking)
+
+                        deviceInitializationTimeoutMillis.set(extension.deviceInitializationTimeoutMillis)
+                        waitForDevicesTimeoutMillis.set(extension.waitForDevicesTimeoutMillis)
+
+                        allureConfiguration.set(extension.allureConfiguration)
+                        timeoutConfiguration.set(extension.timeoutConfiguration)
+                        fileSyncConfiguration.set(extension.fileSyncConfiguration)
+
+                        //Android specific for now
+                        autoGrantPermission.set(extension.autoGrantPermission)
+                        instrumentationArgs.set(extension.instrumentationArgs)
+
+
+                        sdk.set(androidExtension.sdkComponents.sdkDirectory)
+                        outputs.upToDateWhen { false }
```

Comment:
nit: I think `outputs.cacheIf { false }` should go in pair with `outputs.upToDateWhen { false }` to prevent upload cached artifacts which would never be reused.